core/src/main/java/kafka/server/share/SharePartitionManager.java

```diff
@@ -626,6 +613,85 @@ void maybeProcessFetchQueue() {
         }
     }
 
+    private SharePartition fetchSharePartition(SharePartitionKey sharePartitionKey) {
+        return partitionCacheMap.computeIfAbsent(sharePartitionKey,
+            k -> {
+                long start = time.hiResClockMs();
+                SharePartition partition = new SharePartition(
+                    sharePartitionKey.groupId,
+                    sharePartitionKey.topicIdPartition,
+                    maxInFlightMessages,
+                    maxDeliveryCount,
+                    recordLockDurationMs,
+                    timer,
+                    time,
+                    persister
+                );
+                this.shareGroupMetrics.partitionLoadTime(start);
+                return partition;
+            });
+    }
+
+    private void maybeCompleteInitializationWithException(
+        SharePartitionKey sharePartitionKey,
+        CompletableFuture<Map<TopicIdPartition, PartitionData>> future,
+        Throwable throwable) {
+        if (throwable instanceof LeaderNotAvailableException) {
+            log.debug("The share partition with key {} is not initialized yet", sharePartitionKey);
+            // Do not process the fetch request for this partition as the leader is not initialized yet.
+            // The fetch request will be retried in the next poll.
+            // TODO: Add the request to delayed fetch purgatory.
+            return;
+        }
+
+        if (throwable instanceof NotLeaderOrFollowerException || throwable instanceof FencedStateEpochException) {
+            log.info("The share partition with key {} is fenced with message: {}", sharePartitionKey, throwable.getMessage());
```

Comment:
nit: I'd remove the "with message". Still makes sense that you're adding the exception message after the colon.

Reply:
Thanks, done.